dnsmasq: add procd interface index tracking

valikkD opened the pull request:
Problem exist when dnsmasq is exclusively bind to particular interface.
After reconfiguring or restarting this interface, its index changes, but
dnsmasq uses the old one. When this problem occurs, dnsmasq does not
listen on the correct interface so DHCP does not work, and clients do not
get an IP address. Procd netdev param can be added to restart dnsmasq when
the interface index is changed.

Signed-off-by: Valentyn Datsko <valikk.d@gmail.com>

dangowrt commented on package/network/services/dnsmasq/files/dnsmasq.init:
```diff
@@ -1156,6 +1156,16 @@ dnsmasq_start()
+	config_get instance_ifc "$cfg" interface
+	if [ -n "$instance_ifc" ]; then
+		network_get_device instance_netdev "$instance_ifc" && {
+			if [ -n "$instance_netdev" ]; then
```
Why not use `&&` here just like in the line above (there maybe `if ...; then ... fi` would be better because it's more than one line).

valikkD replied:
Okay, I'll replace `&&` with `if`.

Commit:
dnsmasq: add procd interface index tracking

Problem exist when dnsmasq is exclusively bind to particular interface.
After reconfiguring or restarting this interface, its index changes, but
dnsmasq uses the old one. When this problem occurs, dnsmasq does not
listen on the correct interface so DHCP does not work, and clients do not
get an IP address. Procd netdev param can be added to restart dnsmasq when
the interface index is changed.

Signed-off-by: Valentyn Datsko <valikk.d@gmail.com>
